edgelet/edgelet-http/src/unix.rs

```diff
@@ -34,14 +37,35 @@ pub fn listener<P: AsRef<Path>>(path: P) -> Result<Incoming, Error> {
         defer! {{ umask(prev); }}
 
         debug!("binding {}...", path.as_ref().display());
+
         let listener = UnixListener::bind(&path)
             .with_context(|_| ErrorKind::Path(path.as_ref().display().to_string()))?;
         debug!("bound {}", path.as_ref().display());
 
         Incoming::Unix(listener)
     } else {
+        // If parent doesn't exist, create it and socket will be created inside.
+        if let Some(parent) = path.as_ref().parent() {
+            if !parent.exists() {
+                fs::create_dir_all(parent).with_context(|err| {
+                    error!("Cannot create directory, error: {}", err);
+                    ErrorKind::Path(path.as_ref().display().to_string())
+                })?;
+            }
+        }
+
         let listener = UnixListener::bind(&path)
             .with_context(|_| ErrorKind::Path(path.as_ref().display().to_string()))?;
+
+        fs::set_permissions(
+            path.as_ref(),
+            fs::Permissions::from_mode(SOCKET_DEFAULT_PERMISSION),
+        )
```

Comment:
This is wrong if this branch is being used to create the mgmt socket.

Reply:
You're right, good catch. So management socket could not be created and reach here?

Should I set the permission instead in workload manager? Or maybe refactor with an enum to discriminate between between mgmt and workload?

Reply:
>So management socket could not be created and reach here?

It's possible that the user sets `listen.management_uri` to `unix:///some/path/to/mgmt.sock`, but one or more components of `/some/path/to` don't exist, so they'll end up falling to this branch. Before your change this would've been an error. With your change it'll succeed but create the socket with too open permissions.

>Should I set the permission instead in workload manager?

Pass in the desired permissions as a parameter of this function.

Reply:
Fixed